owlplug-client/src/main/java/com/owlplug/project/tasks/discovery/studioone/StudioOneProjectExplorer.java

```diff
@@ -0,0 +1,238 @@
+
+      // Set file timestamps
+      project.setLastModifiedAt(new Date(file.lastModified()));
+      BasicFileAttributes attr = Files.readAttributes(file.toPath(), BasicFileAttributes.class);
+      FileTime fileTime = attr.creationTime();
+      project.setCreatedAt(Date.from(fileTime.toInstant()));
+
+      collectPlugins(tempDir, project);
+
+      return project;
+
+    } catch (IOException e) {
+      throw new ProjectExplorerException(
+          "Error while extracting Studio One project file: " + file.getAbsolutePath(), e);
+    } catch (ParserConfigurationException | SAXException e) {
+      throw new ProjectExplorerException("Error while parsing XML in Studio One project: " + file.getAbsolutePath(), e);
+    } finally {
+      // Clean up temporary directory
+      if (tempDir != null) {
+        try {
+          org.apache.commons.io.FileUtils.deleteDirectory(tempDir.toFile());
+          log.debug("Cleaned up temporary directory: {}", tempDir);
+        } catch (IOException e) {
+          log.debug("Failed to clean up temporary directory: {}", tempDir, e);
+        }
+      }
+    }
+  }
+
+  private void extractMetadata(Document metainfoDoc, XPath xpath, DawProject project) {
+    try {
+      NodeList titleNodes = (NodeList) xpath.compile("//Attribute[@id='Document:Title']/@value")
+              .evaluate(metainfoDoc, XPathConstants.NODESET);
+      if (titleNodes.getLength() > 0) {
+        String title = titleNodes.item(0).getNodeValue();
+        if (title != null && !title.isEmpty()) {
+          project.setName(title);
+        }
+      }
+
+      NodeList creatorNodes = (NodeList) xpath.compile("//Attribute[@id='Document:Creator']/@value")
+              .evaluate(metainfoDoc, XPathConstants.NODESET);
+      if (creatorNodes.getLength() > 0) {
+        String creator = creatorNodes.item(0).getNodeValue();
+        project.setAppFullName(creator != null ? creator : "Studio One");
+      } else {
+        // Try to get from Generator attribute
+        NodeList generatorNodes = (NodeList) xpath.compile("//Attribute[@id='Document:Generator']/@value")
+                .evaluate(metainfoDoc, XPathConstants.NODESET);
+        if (generatorNodes.getLength() > 0) {
+          String generator = generatorNodes.item(0).getNodeValue();
+          if (generator != null && generator.startsWith("Studio One/")) {
+            project.setAppFullName(generator);
+          } else {
+            project.setAppFullName("Studio One");
+          }
+        } else {
+          project.setAppFullName("Studio One");
+        }
+      }
+
+      NodeList versionNodes = (NodeList) xpath.compile("//Attribute[@id='Document:FormatVersion']/@value")
+              .evaluate(metainfoDoc, XPathConstants.NODESET);
+      if (versionNodes.getLength() > 0) {
+        String version = versionNodes.item(0).getNodeValue();
+        project.setFormatVersion(version);
+      }
+    } catch (XPathExpressionException e) {
```

Comment:
We should catch the `XPathExpressionException` for each XPath query here and set a default value accordingly for the `DAWProject`